### ISSUE TITLE
Feature request: Get fit from half-life calculation

### ISSUE THREAD

**Issue by billdenney** [humanpred/pknca#342]
As discussed in #339, there is a need to get a fit from the half-life calculations. @roninsightrx and @kylebaron contributed to that conversation.

In that discussion, there was a desire to have two different things:

1. Return the half-life fit more simply. Specifically, return the slope (`-lambda.z`) and intercept (`log(clast.pred) + tlast * lambda.z,`) for each individual.
2. Enable interpolation or extrapolation of the half-life curve for graphing or other reasons.

@roninsightrx and @kylebaron: are there any other features that you would want?

I think that the first would be easiest by adding one more parameter (maybe named `lambda.z_intercept`) to half-life calculations. (We should double-check to see if there is a CDISC controlled vocabulary name for this parameter before setting the name.)

For the second, I think that doing something like creating an `approx_half_life()` function may be the best (`approx` chosen to match the base R `approx()` function). It would default to something like:

1. Don't extrapolate to earlier times than the first measure included (option to allow extrapolation earlier)
2. Do extrapolate to later times than tlast (option to disable extrapolation later)
3. Exponentiate values so that the result is `concentration` not `log(concentration)` (no option to turn that off, take the log manually if you need it)
4. Optional `tout` (time for output) argument or `n` argument that would do the same as `xout` and `n` in `approx()`

Any other things you think would be helpful?

**Comment by roninsightrx** [comment]
That is a good writeup, and thanks for considering this @billdenney.
1. Agree adding the `lambda.z_intercept` would easiest way to solve this.
2. Agree that a function would be great to have. However, would consider using a different name for it? I actually was not familiar with the base `approx()` function, I guess I would use `predict()` if I had the model or write a few custom lines for it. I was surprised by the name (as it's not really "approximating" anything, just interpolating). Maybe that's just my narrow R knowledge, but a function name like `get_fit_curve()` would perhaps be more intuitive? The argument defaults seem fine to me.

**Comment by billdenney** [comment]
As tangentially discussed in #421, this should not use `tlast` because that could be excluded from the half-life. It should use either `lambda.z.time.first` or `lambda.z.time.last` to ensure that it is going through the points used for calculation.

**Comment by Gero1999** [comment]
Hello @billdenney, would it make sense if we just use the fit coefficient from `half.life.R` to get the intercept? I am just concerned that if we try to extrapolate it using a point it could be the case that the line does not go through it.  I can take care of this if you want and even help with the plotting function, as in `aNCA` we already have something implemented for this.

**Comment by billdenney** [comment]
@Gero1999, Since we should have all the right information without the intercept and the intercept is not an NCA parameter (that I have seen reported or that I see in a quick look of the SDTM parameters), I'd prefer not to add it. I do understand the concern about the possibility of extrapolating from the wrong point, but I fall toward consider one more thing rather than track one more thing.

If the intercept is an SDTM NCA parameter, then I would reconsider in favor of the intercept as it would make PKNCA more SDTM-complete at the same time.

**Comment by Gero1999** [comment]
@billdenney No, unfortunately is not a standard CDISC parameter...Hmm then I wonder, if we think the intercept per ce is not interesting for the user should we:

1) Create a function `get_lambda_z_intercept()` that returns values for each interval
2) Directly do the plotting function recalculating the fit within it (like we do in `aNCA`)

Even if we do not use the original fit, I would still elude extrapolations. We can just use `get_lambda_z_points()` and remake the fit to get the exact coefficient value
